src/secret_base.c

```diff
@@ -2,10 +2,831 @@
 #include "string_util.h"
 #include "strings.h"
 #include "text.h"
+#include "event_data.h"
+#include "vars.h"
+#include "rom4.h"
+#include "asm.h"
+#include "script.h"
+#include "field_player_avatar.h"
+#include "field_camera.h"
+#include "map_constants.h"
+#include "task.h"
+#include "palette.h"
+#include "decoration.h"
+#include "field_weather.h"
+#include "metatile_behavior.h"
+#include "pokemon.h"
 
-u8 sub_80BB8A8(void *);
+extern u8 gUnknown_020387DC;
+extern u16 gSpecialVar_0x8004;
+extern u16 gSpecialVar_0x8005;
+extern u16 gSpecialVar_0x8006;
+extern u16 gSpecialVar_0x8007;
+extern u16 gScriptResult;
+extern const struct
+{
+    u16 unk_083D1358_0;
+    u16 unk_083D1358_1;
+} gUnknown_083D1358[7];
+extern const u8 gUnknown_083D1374[4 * 16];
+extern void *gUnknown_0300485C;
+extern u8 sub_807D770(void);
+extern u8 gUnknown_083D13EC[12];
```

Comment:
Please declare this as `const` along with any other variables that are stored in ROM (ROM variables have an 08XXXXXX address).